components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthUtil.java

```diff
@@ -1059,7 +1074,55 @@ public static boolean revokeTokens(String username, UserStoreManager userStoreMa
         }
 
         boolean isErrorOnRevokingTokens;
+
+        // This is for FEDERATED users from sub organization.
+        String userResidentTenant = null;
+        String authenticatedUserName = null;
+        if (authenticatedUser.getUserResidentOrganization() != null) {
+            try {
+                userResidentTenant = OAuthComponentServiceHolder.getInstance().getOrganizationManager()
+                        .resolveTenantDomain(authenticatedUser.getUserResidentOrganization());
+                if (OrganizationManagementUtil.isOrganization(userResidentTenant)) {
+                    /*
+                     Keeping the username to be used to extract the tokens which are created directly from the
+                     sub organization applications.
+                    */
+                    authenticatedUserName = authenticatedUser.getUserName();
+                    authenticatedUser.setUserName(authenticatedUser.getUserId());
+                }
+            } catch (OrganizationManagementException | UserIdNotFoundException e) {
+                throw new UserStoreException("Error occurred while constructing the authenticated user.", e);
+            }
+        }
         isErrorOnRevokingTokens = processTokenRevocation(clientIds, authenticatedUser, userStoreDomain, username);
+        if (authenticatedUser.getUserResidentOrganization() != null) {
+            /*
+             If the authenticated user's resident organization is an organization, then we need to set the
+             user's username to the user id to get the FEDERATED access tokens issued for the user when
+             accessing the sub organization.
+            */
+            try {
+                /*
+                 If the authenticated user's resident organization is an organization, then we need to check
+                 for the access tokens issued directly for the organization as well.
+                */
+                if (StringUtils.isNotEmpty(userResidentTenant) &&
+                        OrganizationManagementUtil.isOrganization(userResidentTenant)) {
+                    AuthenticatedUser orgUser = new AuthenticatedUser();
+                    orgUser.setUserName(authenticatedUserName);
+                    orgUser.setUserResidentOrganization(authenticatedUser.getUserResidentOrganization());
+                    orgUser.setAccessingOrganization(authenticatedUser.getUserResidentOrganization());
+                    orgUser.setFederatedUser(false);
+                    orgUser.setUserStoreDomain("PRIMARY");
```

Comment:
Why this is hard coded? Can't we get it from authenticatedUser?
Since sub orgs also have secondry userstore support it's possible to log a user from secondary userstore as well

Reply:
Changed